### PR TITLE
Fix named arguments comments

### PR DESCRIPTION
**Pull request by czosel**
I finally figured out what @ichiriac meant by looking ahead in https://github.com/glayzzle/php-parser/pull/673#discussion_r590342168 :relieved: 

cc @cseufert

### CODE REVIEW

**Comment by ichiriac** [comment]
hi @czosel, I didn't had much time to check github messages, I'll be more active in few weeks. I'll take a look of your code and the last commit today.

File: src/parser/function.js [hunk 1]
```diff
@@ -294,14 +294,15 @@ module.exports = {
       this.token === this.tok.T_STRING ||
       Object.values(this.lexer.keywords).includes(this.token)
     ) {
-      const backup = [this.token, this.lexer.getState()];
-      const name = this.text();
-      this.next();
-      if (this.token === ":") {
-        return this.node("namedargument")(name, this.next().read_expr());
+      const lexerState = this.lexer.getState();
+      const nextToken = this.lexer.lex();
```
Comment (ichiriac) [inline src/parser/function.js:298]: 👍

File: test/snapshot/call.test.js [hunk 2]
```diff
@@ -253,6 +253,15 @@ describe("Test call", function () {
     });
     expect(astErr).toMatchSnapshot();
   });
+  it("doesnt confused static methods with named arguments", function () {
+    const astErr = parser.parseEval(`foo(a::bar());`, {
+      parser: {
+        version: "8.0",
```
Comment (ichiriac) [inline test/snapshot/call.test.js:259]: should be interesting to see behavior with `version: "7.2"`

**Review by ichiriac — COMMENTED** [review]
perfect @czosel that's exactly how I would implement it

File: test/snapshot/call.test.js (hunk 2 above)
Comment (czosel) [reply test/snapshot/call.test.js:259]: Good point! I'll send a PR with a test.

**Comment by czosel** [comment]
> hi @czosel, I didn't had much time to check github messages, I'll be more active in few weeks. I'll take a look of your code and the last commit today.

Hi @ichiriac, no worries at all!

When you do find the time again, it would we great if you could take a look at the other commits in the `php8` branch as well (@cseufert sent several PRs).

File: test/snapshot/call.test.js (hunk 2 above)
Comment (czosel) [reply test/snapshot/call.test.js:269]: done! I implemented the version check and added a test.

**Comment by ichiriac** [comment]
@czosel I'll do it an release it under 3.1 when php8 will be fully implemented

**Pull request merged**